### PR TITLE
[rush-lib] Enhance telemetry with individual operation times, graph, and machine info

### PR DESCRIPTION
**Pull request by dmichon-msft**
## Summary
Adds additional information to the standard rush telemetry files, namely detailed machine architecture info (core count, OS, CPU model, RAM), and the operation graph.

Carve out from #3589 to separate concerns.

## Details
Example output from `rush retest -t eslint-plugin`:
```json
[
  {
    "name": "retest",
    "durationInSeconds": 14.688612599968911,
    "result": "Succeeded",
    "extraData": {
      "command_from": "false",
      "command_impactedBy": "false",
      "command_impactedByExcept": "false",
      "command_only": "false",
      "command_to": "true",
      "command_toExcept": "false",
      "command_fromVersionPolicy": "false",
      "command_toVersionPolicy": "false",
      "--timeline": "false",
      "--to": "eslint-plugin",
      "--to-except": "",
      "--from": "",
      "--only": "",
      "--impacted-by": "",
      "--impacted-by-except": "",
      "--to-version-policy": "",
      "--from-version-policy": "",
      "--verbose": "true",
      "--ignore-hooks": "false",
      "--no-color": "false",
      "--update-snapshots": "false",
      "--production": "false",
      "isWatch": false,
      "isInitial": true,
      "countAll": 4,
      "countSuccess": 4,
      "countSuccessWithWarnings": 0,
      "countFailure": 0,
      "countBlocked": 0,
      "countFromCache": 0,
      "countSkipped": 0,
      "countNoOp": 0
    },
    "operationResults": {
      "@rushstack/eslint-plugin (build)": {
        "startSeconds": 3.781,
        "endSeconds": 11.432,
        "result": "SUCCESS",
        "dependencies": [
          "@rushstack/tree-pattern (build)"
        ]
      },
      "@rushstack/tree-pattern (build)": {
        "startSeconds": 0.426,
        "endSeconds": 3.779,
        "result": "SUCCESS",
        "dependencies": []
      },
      "@rushstack/eslint-plugin (test)": {
        "startSeconds": 11.432,
        "endSeconds": 14.982,
        "result": "SUCCESS",
        "dependencies": [
          "@rushstack/eslint-plugin (build)"
        ]
      },
      "@rushstack/tree-pattern (test)": {
        "startSeconds": 3.78,
        "endSeconds": 4.747,
        "result": "SUCCESS",
        "dependencies": [
          "@rushstack/tree-pattern (build)"
        ]
      }
    },
    "machineInfo": {
      "machineArchitecture": "x64",
      "machineCPU": "Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz",
      "machineCores": 8,
      "machineTotalMemoryMiB": 16001,
      "machineFreeMemoryMiB": 11995
    },
    "timestamp": 1661295823893,
    "platform": "linux",
    "rushVersion": "5.76.1"
  }
]
```

## How it was tested
Enabled telemetry in the rushstack repo and ran various commands to test.

### CODE REVIEW

File: libraries/rush-lib/src/logic/Telemetry.ts [hunk 1]
```diff
@@ -1,13 +1,65 @@
...
+  /**
+   * The names of operations that this operation depends on.
+   */
+  dependencies: string[];
+  /**
+   * The status code for the operation.
+   */
+  result: string;
+  /**
+   * A second-resolution timestamp when the operation started.
```
Comment (iclanton) [inline libraries/rush-lib/src/logic/Telemetry.ts:52]: So is this a UNIX timestamp, but in seconds? Why not just use a normal UNIX timestamp?

File: common/changes/@microsoft/rush/enhanced-telemetry_2022-08-23-22-57.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/rush",
+      "comment": "Update telemetry to machine architecture information, and phased command telemetry to have details of the dependency graph and individual build times and statuses.",
```
Comment (iclanton) [inline common/changes/@microsoft/rush/enhanced-telemetry_2022-08-23-22-57.json:5]:
```suggestion
      "comment": "Add machine architecture information, dependency graph information, and individual build times and statuses to the telemetry file.",
```

**Review by iclanton — APPROVED** [review]

File: libraries/rush-lib/src/logic/Telemetry.ts (hunk 1 above)
Comment (dmichon-msft) [reply libraries/rush-lib/src/logic/Telemetry.ts:52]: I can switch it back to milliseconds, but it's not a UNIX timestamp. It's 0 value is when the process started.

**Review by iclanton — APPROVED** [review]

**Comment by hanksterhan** [comment]
Small documentation detail just in case others reference this PR - the `ITelemetryOperationResults` uses the fields `startTimestamp` and `endTimestamp` instead of `startSeconds` and `endSeconds`.

**Comment by dmichon-msft** [comment]
Updated the example text to match the file.

**Comment by octogonz** [comment]
🚀 This was released with Rush 5.77.0

**Pull request merged**